### PR TITLE
EIP-7742: Add target_blob_count to block header

### PR DESCRIPTION
**Pull request by siladu**
## PR description

First major piece of EIP-7742: Add target_blob_count to the block header.

Still todo in future PRs:
- Engine API
- Wiring engine API to block header
- block validation
- gas accounting 

## Fixed Issue(s)

Part of https://github.com/hyperledger/besu/issues/7605

### Thanks for sending a pull request! Have you done the following?

- [x] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [x] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [x] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [x] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [x] integration tests: `./gradlew integrationTest`
- [x] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/jsonrpc/ExecutionEnginePragueAcceptanceTest.java [hunk 1]
```diff
@@ -20,8 +20,11 @@
 
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.params.provider.Arguments;
 
+// TODO SLD
+@Disabled("TODO SLD - Enable when Prague spec is finalized")
```
Comment (siladu) [inline acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/jsonrpc/ExecutionEnginePragueAcceptanceTest.java:27]: I propose we leave this disabled until the nearer the end of the Prague dev work. Anything that changes the block hash (genesis, block header, some engine API changes) causes a cascading change to all the JSON files and it's time consuming to update this with every PR. In the meantime, we will still have regression signal in the devnets and execution-spec/references tests.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/ProcessableBlockHeader.java [hunk 2]
```diff
@@ -46,6 +47,8 @@ public class ProcessableBlockHeader
   protected final Bytes32 mixHashOrPrevRandao;
   // parentBeaconBlockRoot is included for Cancun
   protected final Bytes32 parentBeaconBlockRoot;
+  // TODO SLD Quantity or UInt64Value<UInt64> instead?
+  protected final UInt64 targetBlobCount;
```
Comment (siladu) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/ProcessableBlockHeader.java:51]: Some other code for similar types use a subclass of BaseUInt256Value (which implements Quantity and UInt256Value). 
Withdrawals already uses UInt64 directly.

I think UInt64 is more direct than the tuweni generic base class construct and avoids yet another wrapper class.

Would like to get more opinions about the benefits of defining our own type, i.e. `TargetBlobCount extends BaseUInt256Value<TargetBlobCount> implements Quantity`

**Review by siladu — COMMENTED** [review]
Added some of my own comments for the reviewer

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/ProcessableBlockHeader.java (hunk 2 above)
Comment (siladu) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/ProcessableBlockHeader.java:51]: Side note: Teku using UInt64 directly, as in this PR https://github.com/Consensys/teku/blob/110ce917f1fec942e2c2664c50a1bc3bb778ee07/ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/schema/PayloadAttributesV4.java#L48

File: acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/jsonrpc/ExecutionEnginePragueAcceptanceTest.java (hunk 1 above)
Comment (jframe) [reply acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/jsonrpc/ExecutionEnginePragueAcceptanceTest.java:27]: I think that's fine, still it's a lot of effort to update. Once the testnets are done though we should re-enable this test

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/ProcessableBlockHeader.java (hunk 2 above)
Comment (jframe) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/ProcessableBlockHeader.java:51]: Fine with using UInt64 directly. Adding the new type doesn't add much value

File: config/src/main/java/org/hyperledger/besu/config/GenesisConfigFile.java [hunk 3]
```diff
@@ -265,6 +265,17 @@ public String getParentBeaconBlockRoot() {
         "0x0000000000000000000000000000000000000000000000000000000000000000");
   }
 
+  /**
+   * Gets target blob count.
+   *
+   * @return the target blob count
+   */
+  public Optional<String> getTargetBlobCount() {
```
[the remark this answers is not in the document]
Comment (siladu) [reply config/src/main/java/org/hyperledger/besu/config/GenesisConfigFile.java:273]: yes, [the EIP ](https://eips.ethereum.org/EIPS/eip-7742) says 
> For a genesis block with no existing parent, the value should be set according to the agreed specification for the target blob count given by that genesis block's protocol rule set.

which I have interpreted as pectra-at-genesis networks specifying an initial targetBlobCount.

[the remark this answers is not in the document]
Comment (jframe) [reply config/src/main/java/org/hyperledger/besu/config/GenesisConfigFile.java:273]: I'm not sure how to interrupt that statement in the EIP, but until we know any different happy to go with this

**Review by jframe — APPROVED** [review]

**Pull request merged**